autodist/kernel/synchronization/compressor.py

```diff
@@ -205,6 +205,104 @@ class HorovodCompressorEF(CompressorEF, HorovodCompressor):  # This works becaus
     """Horovod's Compression but with Error Feedback."""
 
 
+class PowerSGDCompressor(CompressorEF):
+    """An implementation of the PowerSGD compression algorithm (arxiv.org/abs/1905.13727)."""
+    def __init__(self, var_op_name, rank=1):
+        self.rank = rank
+        self.og_shape, self.ndims, self.compressor = None, None, None  # compressor is the Q in paper
+        self.var_op_name = var_op_name
+        super.__init__(var_op_name)
+
+    def reduce(self, tensor: Tensor, conf: CollectiveOpsConfig):
+        """
+        Compress, reduce, and decompress a given tensor.
+
+        Args:
+            tensor (Tensor): the Tensor to reduce.
+            conf (CollectiveOpsConfig): the config for Collective Ops.
+
+        Returns:
+            Reduced Tensor
+        """
+        if self.og_shape is None:
+            self.og_shape = tensor.shape
+            self.ndims = len(self.og_shape)
+
+        # rank <= 1
+        if self.ndims <= 1 or (self.ndims==2 and any([d == 1 for d in self.og_shape])):
+            return self._all_reduce(tensor, conf)
+
+        # compressor init
+        if self.compressor is None:
+            self.compressor = random_ops.random_normal([array_ops.shape_v2(tensor)[1], self.rank])
```

Comment:
it's been a while, so I could be wrong, but I think we might have to set a random seed before the creation of the compressor matrix. Otherwise we could end up with different initial compressor matrices across workers

Reply:
The influence of it might exit at the first several iterations. But it's necessary to remove the differences. I set the random seed now. Thanks!